vnext/Desktop/Modules/WebSocketModule.cpp

```diff
@@ -150,12 +149,29 @@ void WebSocketModule::SendEvent(string &&eventName, dynamic &&args) {
 }
 
 // clang-format off
-std::shared_ptr<IWebSocketResource> WebSocketModule::GetOrCreateWebSocket(int64_t id, string&& url)
+shared_ptr<IWebSocketResource> WebSocketModule::GetOrCreateWebSocket(int64_t id, string&& url)
 {
   auto itr = m_webSockets.find(id);
   if (itr == m_webSockets.end())
   {
-    auto ws = m_resourceFactory(std::move(url), /*legacyImplementation*/ false, /*acceptSelfSigned*/ false);
+    shared_ptr<IWebSocketResource> ws;
+    try
+    {
+      ws = m_resourceFactory(std::move(url));
+    }
+    catch (const winrt::hresult_error& e)
+    {
+      SendEvent("webSocketFailed", dynamic::object("id", id)("message", Utf16ToUtf8(e.message())));
+
+      return nullptr;
+    }
+    catch (const std::exception& e)
+    {
+      SendEvent("webSocketFailed", dynamic::object("id", id)("message", e.what()));
+
+      return nullptr;
+    }
```

Comment:
>} [](start = 4, length = 1)

are we guaranteed that no other exception types could be thrown? Should we add catch(...)?

Reply:
Don't know about such guarantee. Adding generic catch.